reactor-core/src/main/java/reactor/core/publisher/SinksSubscribers.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2011-Present VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+
+import org.reactivestreams.Subscription;
+
+import reactor.core.CoreSubscriber;
+import reactor.core.Disposable;
+import reactor.util.context.Context;
+
+/**
+ * @author Simon Baslé
+ */
+final class SinksSubscribers {
+
+	private SinksSubscribers() {}
+
+	static abstract class AbstractSubscriberAdapter<T, SINK> implements CoreSubscriber<T>, Disposable {
+
+		final SINK sink;
+		final Context context;
+		final Sinks.EmitFailureHandler emitFailureHandler;
+
+		volatile Subscription                                                     s;
+		@SuppressWarnings("rawtypes")
+		static   AtomicReferenceFieldUpdater<AbstractSubscriberAdapter, Subscription> S
+				= AtomicReferenceFieldUpdater.newUpdater(AbstractSubscriberAdapter.class, Subscription.class, "s");
+
+		AbstractSubscriberAdapter(SINK sink) {
+			this.sink = sink;
+			this.context = sink instanceof ContextHolder ? ((ContextHolder) sink).currentContext() : Context.empty();
+			this.emitFailureHandler = Sinks.EmitFailureHandler.FAIL_FAST;
+		}
+
+		@Override
+		public Context currentContext() {
+			return context;
+		}
+
+		@Override
+		public void onSubscribe(Subscription s) {
+			if (Operators.setOnce(S, this, s)) {
+				s.request(Long.MAX_VALUE);
+			}
+		}
+
+		@Override
+		public boolean isDisposed() {
+			return s == Operators.cancelledSubscription();
+		}
+
+		@Override
+		public void dispose() {
+			Operators.terminate(S, this);
+		}
+	}
+
+	static final class ManySubscriberAdapter<T> extends AbstractSubscriberAdapter<T, Sinks.Many<T>> {
```

Comment:
IMO, in case of `Many`, we should not request `MAX_VALUE` but rather respect the downstream requests

This is also the reason why I thought that it should be something like `Many#createSubscriber()` or maybe even (yeah, yeah, I know I said that we don't need it before :D) `Many#toProcessor` or similar

Reply:
IIRC none of the currently available processors directly translate the downstream subscribers demand, though `EmitterProcessor` does use its `bufferSize` as a prefetch request.

I don't think we can offer a perfectly generic implementation over arbitrary `Sinks.Many`, and putting a conversion method in the `Sinks.Many` interface is IMO falling back into the old mistake of trying to make too many (false) promises.

The goal of this PR is more to provide a helpful yet simple implementation as an alternative to rolling everything on one's own for basic needs where this approach is enough. For more advanced needs, users would have to consider implementing a processor, or use the `tryEmit` APIs on their own.